tensorflow/core/tpu/tpu_initializer_helper.cc

```diff
@@ -166,7 +181,93 @@ bool TryAcquireTpuLock() {
 
   return should_load_library;
 }
+#if defined(PLATFORM_GOOGLE)
+Status InitializeTpuLibrary(void* library_handle) {
+  return errors::Unimplemented("You must statically link in a TPU library.");
+}
+#else  // PLATFORM_GOOGLE
+#include "tensorflow/core/tpu/tpu_library_init_fns.inc"
+
+Status InitializeTpuLibrary(void* library_handle) {
+  Status s = InitializeTpuStructFns(library_handle);
+
+  // Retrieve arguments from environment if applicable
+  std::pair<std::vector<std::string>, std::vector<const char*> > args =
+      GetLibTpuInitArguments();
+
+  // TPU platform registration must only be performed after the library is
+  // loaded. We do not want to register a TPU platform in XLA without the
+  // supporting library providing the necessary APIs.
+  if (s.ok()) {
+    void (*initialize_fn)(bool init_library, int num_args, const char** args);
+    initialize_fn = reinterpret_cast<decltype(initialize_fn)>(
+        dlsym(library_handle, "TfTpu_Initialize"));
+    (*initialize_fn)(/*init_library=*/true, args.second.size(),
+                     args.second.data());
+
+    RegisterTpuPlatform();
+  }
+
+  return s;
+}
+
+void* CreateGcsFilesystemFn() {
```

Comment:
Mark this function and the below one `static`, since they aren't declared in the header file. See https://google.github.io/styleguide/cppguide.html#Internal_Linkage